src/backend/tools/tavily.py

```diff
@@ -12,26 +14,79 @@ class TavilyInternetSearch(BaseTool):
 
     def __init__(self):
         self.client = TavilyClient(api_key=self.tavily_api_key)
+        self.num_results = 6
 
     @classmethod
     def is_available(cls) -> bool:
         return cls.tavily_api_key is not None
 
     def call(self, parameters: dict, **kwargs: Any) -> List[Dict[str, Any]]:
         query = parameters.get("query", "")
-        content = self.client.search(query=query, search_depth="advanced")
+        result = self.client.search(
+            query=query, search_depth="advanced", include_raw_content=True
```

Comment:
scraping can be done with `include_raw_content` which is tad simpler to implement. alternatively I tried it with beautifulsoup which works just as well and provides a bit more flexibility on how the page is parsed. ultimately decided this will suffice.